src/core/Akka/Actor/Deployer.cs

```diff
@@ -100,11 +100,20 @@ private RouterConfig CreateRouterConfig(string routerTypeAlias, Config deploymen
 
             var path = string.Format("akka.actor.router.type-mapping.{0}", routerTypeAlias);
             var routerTypeName = _settings.Config.GetString(path);
-            var routerType = Type.GetType(routerTypeName);
-            Debug.Assert(routerType != null, "routerType != null");
-            var routerConfig = (RouterConfig)Activator.CreateInstance(routerType, deployment);
 
-            return routerConfig;
+            try
+            {
+                var routerType = Type.GetType(routerTypeName, true);
+                Debug.Assert(routerType != null, "routerType != null");
```

Comment:
The debug.assert calls can be removed as they will never be hit as we are throwing before the line is executed.